embedchain/loaders/repo_loader.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import os
+import shutil
+import subprocess
+import tempfile
+import urllib.parse
+
+from repo_loader import repo_loader
+
+from embedchain.loaders.base_loader import BaseLoader
+
+
+class RepoLoader(BaseLoader):
+    def load_data(self, content):
+        """Load data from a git repository."""
+
+        # Check if content is a local directory
+        if os.path.isdir(content):
+            directory = os.path.abspath(content)
+            logging.debug(f"Loading repository from local directory: {directory}")
+            origin = directory
+        else:
+            # Check if content is a valid URL
+            try:
+                result = urllib.parse.urlparse(content)
+                if all([result.scheme, result.netloc]):
+                    # If content is a valid URL, clone the repository into a temporary directory
+                    temp_dir = tempfile.mkdtemp()
+                    try:
+                        subprocess.run(["git", "clone", content, temp_dir], check=True)
+                        directory = temp_dir
+                        logging.debug(f"Cloned repository from {content} to temporary directory {temp_dir}")
+                        origin = content
+                    except subprocess.CalledProcessError:
+                        shutil.rmtree(temp_dir)  # clean up
+                        raise ValueError(f"Failed to clone repository from URL: {content}")
+                else:
+                    raise ValueError("The content must be a valid local directory or a valid URL")
+            except ValueError as ve:
+                logging.error(str(ve))
+                raise
+
+        # This creates a temporary file and opens it for writing
+        temp_file = tempfile.NamedTemporaryFile(delete=False)
+
+        try:
+            outfile = temp_file.name
+            logging.debug(f"repository temp file located at: {temp_file.name}")
+
+            repo_loader.load(directory, out_path=outfile, preamble="", quiet=True, progress=True)
+
+            # Ensure everything is written
+            temp_file.flush()
+
+            # Seek back to the beginning of the file if you want to read it
+            temp_file.seek(0)
+
+            filecontent = temp_file.read().decode()
+
+            # Files is read again with the readlines method, to count the lines.
+            # In some cases, this might be too expensive just for logging.
+            temp_file.seek(0)
+            lines = temp_file.readlines()
+
+            individual_file_content = filecontent.split("----!@#$----")
```

Comment:
didnt get this.

Reply:
the repo loader creates one giant file. Line 65 is the file separator used by the repo loader.

This splits it back into each file, I think it's better to do "natural" chunking first, before you chunk by length.